File: src/main/java/com/example/stagetest/Stage.java
package com.example.stagetest;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Stage {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String sujet;

    private LocalDate dateDebut;

    private LocalDate dateFin;

    @Enumerated(EnumType.STRING)
    private Statut statut;

    @Enumerated(EnumType.STRING)
    private TypeStage typeStage;

    private String etudiantNom;  // Nom de l'étudiant récupéré via `User-Service`

    private String encadrantNom; // Nom de l'encadrant récupéré via `User-Service`

    private String nomEntreprise;
    public enum Statut{
        AVAILABLE, CLOSED
    }

    public enum TypeStage{
        COMPANY_IMMERSION_INTERNSHIP, FINAL_YEAR_PROJECT
    }

    @ElementCollection
    @Enumerated(EnumType.STRING)
    private List<Technologie> technologies = new ArrayList<>();
    public enum Technologie {
        JAVA, PYTHON, JAVASCRIPT, SPRING, ANGULAR , DOTNET,
    }
    private String description;
}

File: src/stage.ts
export enum Statut {
  AVAILABLE = 'AVAILABLE', // Correspond à l'énumération Java
  CLOSED = 'CLOSED'
}

export enum TypeStage {
  COMPANY_IMMERSION_INTERNSHIP = 'COMPANY_IMMERSION_INTERNSHIP', // Correspond à l'énumération Java
  FINAL_YEAR_PROJECT = 'FINAL_YEAR_PROJECT'
}

export enum Technologie {
  JAVA = 'JAVA',
  PYTHON = 'PYTHON',
  JAVASCRIPT = 'JAVASCRIPT',
  SPRING = 'SPRING',
  ANGULAR = 'ANGULAR',
  DOTNET = 'DOTNET'
}

export class Stage {
  id?: number; // Optionnel pour éviter les erreurs d'initialisation
  sujet: string = '';
  dateDebut: string = ''; // Stocké en format `YYYY-MM-DD`
  dateFin: string = '';
  statut: Statut = Statut.AVAILABLE; // Valeur par défaut
  etudiantNom: string = ''; // Récupéré via User-Service
  encadrantNom: string = ''; // Récupéré via User-Service
  nomEntreprise: string = '';
  typeStage: TypeStage = TypeStage.COMPANY_IMMERSION_INTERNSHIP; // Valeur par défaut
  technologies: Technologie[] = []; // Liste des technologies, correspond à l'ElementCollection dans Spring Boot
  description: string = '';
}
